how-to-build-gcc-from-source.rst

```diff
@@ -0,0 +1,133 @@
+How to Build GCC from Source
+============================
+
+Step 1: Install Dependancies
+----------------------------
+
+.. code-block:: console
+
+   sudo dnf install gcc-c++ \
+	libmpc-devel \
+	mpfr-devel \
+	gmp-devel \
+	bison \
+	flex \
+	dejagnu \
+	zlib-devel \
+	glibc-devel.i686
+
+Step 2: Set up Directories
+--------------------------
+
+Development for GCC is split into two different directories: A source, and 
+a build. The source directory is where all the source files will go. This is
+where you will clone the git repo, and where you will be changing files for 
+development. The build directory is where you are going to configure and
+build gcc. This directory contains all of the object files. After every
+edit to the source directory, you will have to recompile the object files.
+
+.. code-block:: console
+
+   mkdir -p ~/bld/trunk && mkdir ~/src/
+
```

Comment:
It's probably worth giving an estimate of the disk space requirements for each of the source and build directories, since gcc is a much larger project than many people are used to dealing with.